### PR TITLE
build: add missing supported engine information

### PR DESCRIPTION
**Pull request by flovogt**
The middleware uses ES Modules which were enabled in node with v15.3.0. I tend to set the minimum requiered version to node 16.

### Dependencies and Development Dependencies

**dependencies**:
- `body-parser`: node >= 0.8 and npm 1.2.8000 || >= 1.4.16
- `istanbul-lib-coverage`: node >=8
- `istanbul-lib-instrument`: node >=8
- `istanbul-lib-report`: node >=10
- `istanbul-reports`: node >=8
- `router`: node >= 0.8
- `serve-static`: node >= 0.8.0
- `xml2js`: node >=4.0.0
  
**devDependencies**:
- `@istanbuljs/esm-loader-hook`: node >=16.12.0
- `ava`: node >=14.19 <15 || >=16.15 <17 || >=18
- `depcheck`: node >=10
- `eslint`: node ^12.22.0 || ^14.17.0 || >=16.0.0
- `eslint-config-google`: node >=0.10.0"
- `eslint-plugin-ava`: node >=14.17 <15 || >=16.4
- `esmock`: node >=14.16.0 <=19.9.0
- `get-port`: node ^12.20.0 || ^14.13.1 || >=16.0.0
- `nyc`: node >=8.9
- `sinon`: no engine defined, but uses es2017 which is fully enabled with node 10
- `supertest`: node >=6.4.0

### CODE REVIEW

**Comment by matz3** [comment]
Thanks for looking up those versions! As the extension requires UI5 Tooling v3 I'd suggest to just also define `^16.18.0 || >=18.12.0`. I don't think we should define versions less than those.

File: package.json [hunk 1]
```diff
@@ -23,6 +23,10 @@
 		"type": "git",
 		"url": "git@github.com:SAP/ui5-tooling-extensions.git"
 	},
+	"engines": {
+		"node": ">=18",
```
Comment (matz3) [inline package.json:27]: Why is this different?

**Review by matz3 — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (flovogt) [reply package.json:27]: npm 9 is not included in node 16

Comment (flovogt) [reply package.json:27]: but sure... some one could install npm manually to v9 and still use node 16

**Review by matz3 — APPROVED** [review]

**Pull request merged**